code_samples/sql-data-types.js

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { Client, SqlColumnType, HzLocalDate, HzLocalDateTime, HzOffsetDateTime, HzLocalTime } =
+    require('hazelcast-client');
+const long = require('long');
+
+// Portable class
+class Student {
+    constructor(age, height) {
+        this.age = age;
+        this.height = height;
+        this.factoryId = 23;
+        this.classId = 1;
+    }
+
+    readPortable(reader) {
+        this.age = reader.readInt('age');
+        this.height = reader.readDouble('height');
+    }
+
+    writePortable(writer) {
+        writer.writeInt('age', this.age);
+        writer.writeDouble('height', this.height);
+    }
+}
+
+const varcharExample = async (client) => {
```

Comment:
I think showing all data types, especially DECIMAL, and DATE/TIME related ones are important. REAL is also important since we are showing a CAST is necessary there. Another option is to just show a couple; but then the user will have some difficulty using sql.

Reply:
Because casting is necessary for some of the types.

Reply:
I believe if the type is not easily usable from Node.js client, we can omit it. My reasoning is that, such types will require Java client access, and the users who know how to use both clients can understand what to do looking at the documentation. Besides, when a user tries to run this code sample, he will see errors instead of outputs and maybe think something is broken

Reply:
I would just put string, portable and number examples and that would probably be enough

Reply:
is current integers example ok? or we should have single integer type?

Reply:
hm ok now I saw your note in parantheses, sorry

Reply:
I would choose the one you need a cast and explain the reasoning